cobigen-cli/cli-systemtest/src/test/java/com/devonfw/cobigen/cli/systemtest/AdaptTemplatesCommandIT.java

```diff
@@ -3,44 +3,134 @@
+    Path templateSetComplex = adaptedTemplateSetsFolderPath.resolve("crud-java-server-app-complex-1.0.0");
+
+    // check if adapted template set exists
+    assertThat(templateSetSimple).exists();
+    assertThat(templateSetComplex).exists();
+
+    Path templateSetResourcesPath = templateSetSimple
+        .resolve(ConfigurationConstants.MAVEN_CONFIGURATION_RESOURCE_FOLDER);
+    Path templateSetResourcesPathComplex = templateSetComplex
+        .resolve(ConfigurationConstants.MAVEN_CONFIGURATION_RESOURCE_FOLDER);
+
+    // check if templates folder exists
+    assertThat(templateSetSimple.resolve(templateSetResourcesPath).resolve(ConfigurationConstants.TEMPLATES_FOLDER))
+        .exists();
+    assertThat(
+        templateSetComplex.resolve(templateSetResourcesPathComplex).resolve(ConfigurationConstants.TEMPLATES_FOLDER))
             .exists();
+
+    // check if template-set.xml exists
+    assertThat(templateSetSimple.resolve(templateSetResourcesPath)
+        .resolve(ConfigurationConstants.TEMPLATE_SET_CONFIG_FILENAME)).exists();
+    assertThat(templateSetComplex.resolve(templateSetResourcesPathComplex)
+        .resolve(ConfigurationConstants.TEMPLATE_SET_CONFIG_FILENAME)).exists();
+
+    // validate correct folder structure
+    assertThat(templateSetSimple.resolve(templateSetResourcesPath)
+        .resolve(ConfigurationConstants.TEMPLATE_SET_FREEMARKER_FUNCTIONS_FILE_NAME)).exists();
+    assertThat(templateSetComplex.resolve(templateSetResourcesPathComplex)
+        .resolve(ConfigurationConstants.TEMPLATE_SET_FREEMARKER_FUNCTIONS_FILE_NAME)).exists();
+
+    // check if template set utility resource folder exists
+    assertThat(templateSetSimple.resolve(ConfigurationConstants.UTIL_RESOURCE_FOLDER)).exists();
+    assertThat(templateSetComplex.resolve(ConfigurationConstants.UTIL_RESOURCE_FOLDER)).exists();
+
+    // validate maven specific contents
+    assertThat(templateSetSimple.resolve("pom.xml")).exists();
+    assertThat(templateSetComplex.resolve("pom.xml")).exists();
+
+    // check if META-INF was deleted
+    assertThat(templateSetResourcesPath.resolve("META-INF")).doesNotExist();
+    assertThat(templateSetResourcesPathComplex.resolve("META-INF")).doesNotExist();
+
+  }
+
+  /**
+   * Checks if adapt-templates command successfully created cobigen templates folder and its sub folders
+   *
+   * @throws Exception test fails
+   */
+  @Test
+  public void adaptTemplatesTest() throws Exception {
+
+    Path templatesPath = this.currentHome.resolve(ConfigurationConstants.TEMPLATES_FOLDER);
+    Path CobigenTemplatesPath = templatesPath.resolve(ConfigurationConstants.COBIGEN_TEMPLATES);
+    if (!Files.exists(templatesPath)) {
+      Files.createDirectories(templatesPath);
+    }
+
+    TemplatesJarUtil.downloadJar("com.devonfw.cobigen", "templates-devon4j", "3.0.0", false, templatesPath.toFile());
+    TemplatesJarUtil.downloadJar("com.devonfw.cobigen", "templates-devon4j", "3.0.0", true, templatesPath.toFile());
```

Comment:
This could be replaced with the new downloadFromUrl method. So we can then also get rid of the downloadJar method completely.